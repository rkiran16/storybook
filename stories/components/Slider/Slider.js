import { html } from "lit-html";
import { TextInput } from "../../form/TextInput/TextInput";

export const Slider = ({range, disabled}) => {
    return html`
        <div class="slider ${disabled ? "disabled" : ""}">
            <label for="rangeSlider" class="form-label">Slider label</label>
            <div class="d-flex align-items-center">
                <p class="left-label">0</p>
                <input type="range" class="form-range" id="rangeSlider">
                <p>100</p>
                ${range ? html`<div class="textInput">${TextInput({label: "Min", placeholderText: "0", disabled: disabled})}</div>
                <div class="textInput">${TextInput({label: "Max", placeholderText: "100", disabled: disabled})}</div>
                ` : html`<div class="textInput">${TextInput({label: "Qty", placeholderText: "0", disabled: disabled})}</div>
                `}
            </div>
        </div>
    `;
}
