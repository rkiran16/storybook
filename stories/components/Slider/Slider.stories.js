import { Slider } from "./Slider";

export default {
    title: "Components/Slider",
    parameters: {
        zeplinLink: "https://app.zeplin.io/project/62902a463701ffb0e532df2b/screen/62aa18a0aa9226306da938b5"
    },
    argTypes: {
        range: {
            name: "Range Slider", 
            type: { name: 'boolean', required: false},
            description: "Show Range Slider",
            defaultValue: false,
            table: {
              type: { summary: 'boolean' },
              defaultValue: { summary: false },
            },
            control: {
              type: 'boolean'
            }
           },
        disabled: {
            name: "Disabled",
            type: { name: 'boolean', required: false},
            description: "Disable Slider",
            defaultValue: false,
            table: {
              type: { summary: 'boolean' },
              defaultValue: { summary: false },
            },
            control: {
              type: 'boolean'
            }
          }
    }
}

const Template = (args) => Slider(args);

export const ValueSlider = Template.bind({});
ValueSlider.args = {
    range: false
}

export const RangeSlider = Template.bind({});
RangeSlider.args = {
    range: true
}
