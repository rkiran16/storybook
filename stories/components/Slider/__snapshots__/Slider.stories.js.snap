// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Components/Slider Range Slider 1`] = `
Object {
  "_$litType$": 1,
  "strings": Array [
    "
        <div class=\\"slider ",
    "\\">
            <label for=\\"rangeSlider\\" class=\\"form-label\\">Slider label</label>
            <div class=\\"d-flex align-items-center\\">
                <p class=\\"left-label\\">0</p>
                <input type=\\"range\\" class=\\"form-range\\" id=\\"rangeSlider\\">
                <p>100</p>
                ",
    "
            </div>
        </div>
    ",
  ],
  "values": Array [
    "",
    Object {
      "_$litType$": 1,
      "strings": Array [
        "<div class=\\"textInput\\">",
        "</div>
                <div class=\\"textInput\\">",
        "</div>
                ",
      ],
      "values": Array [
        Object {
          "_$litType$": 1,
          "strings": Array [
            "
      <div class=\\"form-group ",
            "\\">
        <input type=\\"text\\"
        class=",
            " 
        ?disabled=",
            " id=\\"floatingInput\\" placeholder=\\"",
            "\\">
        <label for=\\"floatingInput\\">",
            "</label>
        ",
            "
        ",
            "
        ",
            "
      </div>
  ",
          ],
          "values": Array [
            "",
            "form-control form-control-undefined ",
            false,
            "0",
            "Min",
            "",
            "",
            "",
          ],
        },
        Object {
          "_$litType$": 1,
          "strings": Array [
            "
      <div class=\\"form-group ",
            "\\">
        <input type=\\"text\\"
        class=",
            " 
        ?disabled=",
            " id=\\"floatingInput\\" placeholder=\\"",
            "\\">
        <label for=\\"floatingInput\\">",
            "</label>
        ",
            "
        ",
            "
        ",
            "
      </div>
  ",
          ],
          "values": Array [
            "",
            "form-control form-control-undefined ",
            false,
            "100",
            "Max",
            "",
            "",
            "",
          ],
        },
      ],
    },
  ],
}
`;

exports[`Storyshots Components/Slider Value Slider 1`] = `
Object {
  "_$litType$": 1,
  "strings": Array [
    "
        <div class=\\"slider ",
    "\\">
            <label for=\\"rangeSlider\\" class=\\"form-label\\">Slider label</label>
            <div class=\\"d-flex align-items-center\\">
                <p class=\\"left-label\\">0</p>
                <input type=\\"range\\" class=\\"form-range\\" id=\\"rangeSlider\\">
                <p>100</p>
                ",
    "
            </div>
        </div>
    ",
  ],
  "values": Array [
    "",
    Object {
      "_$litType$": 1,
      "strings": Array [
        "<div class=\\"textInput\\">",
        "</div>
                ",
      ],
      "values": Array [
        Object {
          "_$litType$": 1,
          "strings": Array [
            "
      <div class=\\"form-group ",
            "\\">
        <input type=\\"text\\"
        class=",
            " 
        ?disabled=",
            " id=\\"floatingInput\\" placeholder=\\"",
            "\\">
        <label for=\\"floatingInput\\">",
            "</label>
        ",
            "
        ",
            "
        ",
            "
      </div>
  ",
          ],
          "values": Array [
            "",
            "form-control form-control-undefined ",
            false,
            "0",
            "Qty",
            "",
            "",
            "",
          ],
        },
      ],
    },
  ],
}
`;
